llm/main.py

```diff
@@ -204,21 +206,19 @@ def main(cfg):
         schedulers=scheduler,
         max_duration=cfg.max_duration,
         eval_interval=cfg.eval_interval,
-        eval_subset_num_batches=cfg.eval_loader.get('eval_subset_num_batches',
-                                                    -1),
+        eval_subset_num_batches=cfg.eval_subset_num_batches,
```

Comment:
Assuming every config using this code will set `eval_subset_num_batches` is a strong assumption.
Most ppl would expect it to be `-1` ie eval on the entire val set.

Reply:
I think this is related to the "where defaults go" issue. I think I'd rather explicitly put it into every YAML as -1, because that way as a new user you can see right away in both `main.py` and in the YAMLs that it's an option. I think we've had a couple customers not be able to find this flag before so I'm trying to make it more visible all throughout

Reply:
I guess I can revert it for now to not break existing YAMLs, and we can make a JIRA ticket to talk about defaults after?